File: Assignment_5/demo/test_a5_demo.py
# pytest test_a5_demo.py
import pytest

import a5_demo


def test_basic_one():
    actual = a5_demo.plus_one(1)
    expected = 2
    assert actual == expected


def test_decimal():
    actual = a5_demo.plus_one(1.234)
    expected = 2.234
    assert actual == expected


def test_negative():
    actual = a5_demo.plus_one(-51)
    expected = -50
    assert actual == expected


def test_large_number_fail():
    actual = a5_demo.plus_one(1e6)
    expected = 1000000
    assert actual == expected

File: Assignment_5/demo/test_a5_demo.R
# testthat::test_file("test_a5_demo.R")
library(testthat)

source("a5_demo.R")

test_that("test_basic", {
  actual <- plus_one(1)
  expected <- 2
  expect_equal(actual, expected)
})

test_that("test_decimal", {
  actual <- plus_one(1.234)
  expected <- 2.234
  expect_equal(actual, expected)
})

test_that("test_negative", {
  actual <- plus_one(-51)
  expected <- -50
  expect_equal(actual, expected)
})

test_that("test_large_number_fail", {
  actual <- plus_one(1e6)
  expected <- 1000000
  expect_equal(actual, expected)
})
